Firestore/Source/API/FIRLocalCacheSettings+Internal.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "FIRLocalCacheSettings.h"
+
+#import <Foundation/Foundation.h>
+
+#include <memory>
+#include "Firestore/core/src/api/settings.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FIRPersistentCacheSettings (Internal)
+
+@property(nonatomic, assign)
+    const firebase::firestore::api::PersistentCacheSettings& internalSettings;
```

Comment:
This is mainly for my benefit (pardon my ignorance. I'm not an obj-c expert).

I was at first a bit confused about having `internalSettings` here as public `@property` and then having `_internalSettings` as a private member in the .mm file.

this results in code like

return **_internalSettings** == otherSettings.**internalSettings**;

IIUC `otherSettings.internalSettings` invokes the getter method, and it's all good. But I wonder if a method is more idiomatic? e.g.

```
@interface FIRPersistentCacheSettings (Internal)
- (const firebase::firestore::api::PersistentCacheSettings&)internalSettings;
@end
```